cpp/src/arrow/filesystem/gcsfs.cc

```diff
@@ -962,7 +962,7 @@ Result<std::shared_ptr<io::OutputStream>> GcsFileSystem::OpenAppendStream(
   return Status::NotImplemented("Append is not supported in GCS");
 }
 
-std::shared_ptr<GcsFileSystem> GcsFileSystem::Make(const GcsOptions& options,
+Result<std::shared_ptr<GcsFileSystem>> GcsFileSystem::Make(const GcsOptions& options,
```

Comment:
So this is for compatible reason? 🤔

Reply:
Yes correct, as per the original issue: "For consistency with S3FileSystem" which, incidentally, makes it consistent with other similar filesystems.

Reply:
Done! Ref b27686b4ccfb49171ca52596ab28734024696dcf (although I had to use 18 as major version number for workflow to finish).